portfolio/src/main/webapp/script.js

```diff
@@ -109,22 +178,164 @@ function retrieveId(string, source){
+    var wordArr = text.split(/[\s,"]+/);
+
+    text = [];
+    wordArr.forEach(function(el) {
+        if(el.includes(wantedWord)){    
+            el = '<span class="highlight" id="spanned">' + el + '</span>';
+        }
+        text.push(el);
+    });
+
+    text = text.join(' ');
+    div.innerHTML = text;
+}
+
+/**
+ *  Reverts given div back to original source code
+ */
+function clearDivAtIndex(div){
+    document.getElementById(div).innerHTML = documentCopy.getElementById(div).innerHTML; 
+}
+
+/**
+ *  Removes duplicates in an array
+ */
+function removeDuplicates(array){
+    newArr = [...new Set(array)];
+    return newArr;
+}
+
+/**
+ * Handles physically highlighting each match and clearing matches prior
+ */
+function runHighlights(ids, highlightedWord, src, index){
+    newids = removeDuplicates(ids);
+
+    //if done with all matches
+    if (index === newids.length){
+        clearDivAtIndex(newids[index-1]);
+        setHidden('nextbutton', true);
+        document.body.scrollIntoView({behavior: "smooth"}); //back to the top
+        return;
+    }
+    
+    //clear the prior div
+    if (index > 0){
+        clearDivAtIndex(newids[index-1]);
+    }
+
+    wrapInSpan(newids[index], highlightedWord);
+
+    // no support for safari or ie
+    document.getElementById(newids[index]).scrollIntoView({behavior: "smooth"}); 
+}
+
 function runSearch(input){
+    
+
+    //show the next button
+    setHidden('nextbutton', false);
+
+    var counter = 0;
     var source = retrieveHtml();
 
-    var foundIndex = source.indexOf(input);
-    if (foundIndex == -1){return};
+    var foundIndices = getIndicesOf(input, source);
+    if (foundIndices.length == 0){
+        setHidden('nextbutton', true);
+        shakeSearchBar();
+        return;
+    }
 
-    var closestTag = getClosestTag(foundIndex, source); 
-    var gottenId = retrieveId(closestTag, source);
+    var tags = getTags(foundIndices, source); 
+    var ids = getIds(tags, source);
 
-    if (gottenId == null){
-        alert("gottenId == null");
+    //if none of the matches have corresponding ids
+    if(ids.length == 0){
+        setHidden('nextbutton', false);
+        alert("no gotten Ids");
         return;
     }
 
-    // no support for safari or ie
-    document.getElementById(gottenId).scrollIntoView({behavior: "smooth"}); 
+    //Enter searching mode
+    var el = document.getElementById("nextbutton")
+    el.addEventListener("click", function(){ 
+        if (counter == 0){
+            document.getElementById(ids[0]).scrollIntoView({behavior: "smooth"});
+        }
+        runHighlights(ids, input, documentCopy, counter);
+        counter++;
+    });
+
+    return; //success
+}
+
+/**
+ * EXPERIMENTAL -- for use in walking the tree
+ */
+function textNodesUnder(node){
+  var all = [];
+  for (node=node.firstChild;node;node=node.nextSibling){
+    if (node.nodeType==3) all.push(node);
+    else all = all.concat(textNodesUnder(node));
```

Comment:
nit: if statements with an else should be separated with braces https://google.github.io/styleguide/jsguide.html#formatting-braces-all